internal/provider/models/processors/test/provider_test.go

```diff
@@ -1,11 +1,42 @@
 package processors
 
 import (
+	"context"
+	"fmt"
+	"time"
+
 	"github.com/hashicorp/terraform-plugin-framework/providerserver"
 	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
 	"github.com/mezmo/terraform-provider-mezmo/internal/provider"
+	"github.com/mezmo/terraform-provider-mezmo/internal/provider/providertest"
 )
 
 var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
 	"mezmo": providerserver.NewProtocol6WithError(provider.New("processors_test")()),
 }
+
+// testAccBackend checks that userConfig matches whats is in the
+// backend for resourceName
```

Comment:
I'd like to understand what use case this solves that `StateHasExpectedValues` doesn't solve?  IMO, we shouldn't be verifying the DB on the backend since we should be trusting what the api responses are (and that they're set in the DB).

`StateHasExpectedValues` is designed to check that the state matches the given values.  This is agnostic of any typing or storage on the BE since the state should always match what we provide in the test.  To me, that solves most/all of the problems with the BE being out of sync or not storing something correctly.

I'm not trying to pick at these approaches, it just feels like we're developing multiple ways of doing the same testing which can get confusing as to what tool to use and why use one over another.

Reply:
I thought the idea behind NullifyPlanFields() was to check that what got into the state is also whats on the backend. Since we're not using that "check" for the file consolidating type I added this as a way to check the backend.

It is also suggested in the docs that a "basic test" should validate that the values got into the backend: https://developer.hashicorp.com/terraform/plugin/testing/testing-patterns#basic-test-to-verify-attributes

Reply:
Ah, I see you responded.  Missed this on the main thread (this PR is pretty huge).

I can appreciate the TF suggestion, but I'm still not sure it's necessary.  It feels like `StateHasExpectedValues` should be able to nullify this custom struct directly, but I haven't looked.  I can't help but feeling like we're doing extra work instead of fixing the tools we have.

Reply:
And yes, that's the intention of `NullifyPlanFields` - everything we have configured should be `.Set()` in the state after an insert/update.  If not, something is off either with our service or the provider.  I haven't seen the need to check the BE when the api responses are giving us the values stored on the server in its response.

Reply:
Do you want me to remove the test case?